src/daemon/XorgDisplayServer.cpp

```diff
@@ -285,11 +285,21 @@ namespace SDDM {
         env.insert(QStringLiteral("PATH"), mainConfig.Users.DefaultPath.get());
         env.insert(QStringLiteral("XAUTHORITY"), m_authPath);
         env.insert(QStringLiteral("SHELL"), QStringLiteral("/bin/sh"));
+        env.insert(QStringLiteral("XCURSOR_THEME"), mainConfig.Theme.CursorTheme.get());
         displayScript->setProcessEnvironment(env);
 
         // delete displayScript on finish
         connect(displayScript, SIGNAL(finished(int,QProcess::ExitStatus)), displayScript, SLOT(deleteLater()));
```

Comment:
This can't be safe.

We're calling deleteLater() when we finish xsetroot  and then starting X with that QProcess